packages/core/admin/admin/src/content-manager/utils/schema.js

```diff
@@ -218,8 +218,25 @@ const createYupSchemaAttribute = (type, validations, options) => {
 
   if (type === 'blocks') {
     schema = yup.mixed().test('isJSON', errorsTrads.json, (value) => {
+      // Disable the test for bulk publish, it's valid when it comes from the db
+      if (options.isJSONTestDisabled) {
+        return true;
+      }
+
+      // Don't run validations on drafts
+      if (options.isDraft) {
+        return true;
+      }
+
+      /**
+       * The value is already a parsed object if the onChange event was never fired (i.e. when publishing).
```

Comment:
I know this works, but could it be possible to ensure that the source of the data is never a parsed object instead? I find it weird to have different types possible based on the context, and also odd to do JSON stringify and parse in a row here

Reply:
Yes I understand the concern and I spent time trying to find another solution. The problem is the slate editor expects a parsed object as a value not a string. The onChange needs a string and stores the value on the modifiedData as a string. An entry with no change received a valid parsed object from the db. I see a couple options:

- Skip the validation onPublish, the data has already been validated after change and onSave
- Stringify the data before publishing in this file by checking the modified data for type blocks:

https://github.com/strapi/strapi/blob/31519d59789e30f89740ed25b52cc9503782ef4e/packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/index.js#L458

- Do what I'm doing here

I chose doing what I'm doing here mostly because it is isolated. Otherwise I have to touch files that have no tests to skip the validation or do more or less what I'm doing here.

I'll spend a bit more time, If you have another idea I'm all ears or I would be happy to pair later to take a look.

Also, what's even the benefit of this validation? I feel like maybe I'm missing something because it seems pretty close to useless.